src/lib/module-info.ts

```diff
@@ -355,18 +359,37 @@ export function getTestDependencies(
                     `'${filePath()}' unnecessarily references '${referencedPackage}', which is already referenced in the type definition.`);
             }
             if (referencedPackage === packageName) {
-                throw new Error(`'${filePath()}' unnecessarily references the package. This can be removed.`);
+                referencesSelf = true;
             }
             testDependencies.add(referencedPackage);
         }
         for (const imported of imports(sourceFile)) {
+            hasImports = true;
             if (!imported.startsWith(".")) {
                 const dep = rootName(imported, typeFiles);
                 if (!dependencies.has(dep) && dep !== packageName) {
                     testDependencies.add(dep);
                 }
             }
         }
+
+        isModule = hasImports || (() => {
+            // FIXME: This results in files without imports to be walked twice,
```

Comment:
good to note, although performance hasn't been a problem yet, and I just got back 35% or so from removing `async`.